src/github.com/getlantern/flashlight/statserver/peer.go

```diff
@@ -97,18 +98,20 @@ func (peer *Peer) geolocate() error {
 	var err error
 
 	for i := 0; i < maxGeolocateTries; i++ {
+
 		if i > 0 {
+			// Maximum sleep time: 2^(maxGeolocateTries - 1) * retryWaitTime
 			retryWait := time.Duration(math.Pow(2, float64(i)) * float64(retryWaitTime))
-			log.Debugf("Waiting %v before retrying geolocate", retryWait)
+			log.Errorf("Failed geolocation attempt %d/%d: %q. Waiting %v before retrying.", i, maxGeolocateTries-1, err, retryWait)
```

Comment:
Instead of the error log at line 111, we log here the number of attempt, the cause of the failure and the time the goroutine is going to sleep.